analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/PubliclyWritableDirectoriesBase.cs

```diff
@@ -102,11 +96,29 @@ protected override void Initialize(SonarAnalysisContext context)
                 Language.SyntaxKind.InvocationExpression);
         }
 
-        private bool IsSensitiveDirectoryUsage(string directory) =>
-            windowsDirectories.IsMatch(directory)
-            || macDirectories.IsMatch(directory)
-            || linuxDirectories.IsMatch(directory)
-            || environmentVariables.IsMatch(directory)
-            || userProfile.IsMatch(directory);
+        private static bool IsSensitiveDirectoryUsage(string directory) =>
+            WindowsDirectories.IsMatch(directory)
+                || MacDirectories.IsMatch(directory)
+                || LinuxDirectories.IsMatch(directory)
+                || EnvironmentVariables.IsMatch(directory)
+                || UserProfile.IsMatch(directory);
+
+        private static string[] LinuxDirs() => new[]
+            {
+                "/dev/mqueue",
+                "/run/lock",
+                "/var/run/lock",
+            };
+
+        private static string[] MacDirs() => new[]
+            {
+                "/var/tmp",
+                "/usr/tmp",
+                "/dev/shm",
+                "/library/caches",
+                "/users/shared",
+                "/private/tmp",
+                "/private/var/tmp",
+            };
```

Comment:
Why switch from fields to methods?
What are the benefits of not making the fields static?

Reply:
They are only ever used in the static ctor. There is no need to keep them in a static field afterward.